include/deal.II/base/data_out_base.h

```diff
@@ -2826,39 +2827,90 @@ class DataOutReader : public DataOutInterface<dim,spacedim>
 
 
 /**
- * A class to store relevant data to use when writing the light data XDMF
- * file. This should only contain valid data on the root node which writes the
- * files, the rest of the nodes will have valid set to false. The XDMF file in
- * turn points to heavy data files (such as HDF5) where the actual simulation
- * data is stored. This allows flexibility in arranging the data, and also
+ * A class to store relevant data to use when writing a lightweight XDMF
+ * file. The XDMF file in turn points to heavy data files (such as HDF5)
+ * where the actual simulation data is stored.
+ * This allows flexibility in arranging the data, and also
  * allows the mesh to be separated from the point data.
  */
 class XDMFEntry
 {
-private:
-  /// Whether this entry is valid and contains data to be written
-  bool                                valid;
-  /// The name of the HDF5 heavy data solution and/or mesh files this entry references
-  std::string                         h5_sol_filename, h5_mesh_filename;
-  /// The simulation time associated with this entry
-  double                              entry_time;
-  /// The number of nodes, cells and dimensionality associated with the data
-  unsigned int                        num_nodes, num_cells, dimension;
-  /// The attributes associated with this entry and their dimension
-  std::map<std::string, unsigned int> attribute_dims;
+public:
+  /**
+   * Default constructor that creates an invalid object.
+   */
+  XDMFEntry()
+    :
+    valid(false),
+    h5_sol_filename(""),
+    h5_mesh_filename(""),
+    entry_time(0.0),
+    num_nodes(numbers::invalid_unsigned_int),
+    num_cells(numbers::invalid_unsigned_int),
+    dimension(numbers::invalid_unsigned_int),
+    space_dimension(numbers::invalid_unsigned_int)
+  {};
```

Comment:
Try out the corresponding functionality of eclipse -- not perfect, but gets you one step closer.